src/main/java/io/github/dlinov/leetcode/Task074SearchA2DMatrix.java

```diff
@@ -0,0 +1,20 @@
+package io.github.dlinov.leetcode;
+
+public class Task074SearchA2DMatrix {
+    class Solution {
+        public boolean searchMatrix(int[][] matrix, int target) {
+            if (matrix.length == 0 || matrix[0].length == 0) return false;
+            final int N = matrix.length * matrix[0].length;
+            return binSearch(matrix, target, 0, N - 1);
```

Comment:
That looks amazing! Could you please also implement a different version of binary search algorithm and set the initial borders as [-1, N]?
![image](https://user-images.githubusercontent.com/5459870/96385148-f5fb2200-1189-11eb-9ef5-f53cc745b2e3.png)